lib/targets.go

```diff
@@ -40,17 +43,128 @@ func (t *Target) Request() (*http.Request, error) {
 	return req, nil
 }
 
+// Equal returns true if the target is equal to the other given target.
+func (t *Target) Equal(other *Target) bool {
+	switch {
+	case t == other:
+		return true
+	case t == nil || other == nil:
+		return false
+	default:
+		equal := t.Method == other.Method &&
```

Comment:
Guess this is for readability, would argue just as hard to read in the if statement itself.